File: pingdart-dart/lib/pingdart_sdk.dart
import 'package:http/http.dart' as http;
import 'src/services/database_service.dart';
import 'src/services/calls_service.dart';
import 'src/services/ai_service.dart';
import 'src/services/email_service.dart';
import 'src/services/sms_service.dart';
import 'src/services/storage_service.dart';

class PingDartSDK {
  late final DatabaseService database;
  late final CallsService calls;
  late final AiService ai;
  late final EmailService email;
  late final SmsService sms;
  late final StorageService storage;

  PingDartSDK({
    required String apiKey,
    String? databaseId,
    String baseUrl = 'https://cloudapi.pingdart.com/api',
    String? realtimeBaseUrl,
  }) {
    final client = http.Client();
    
    final rUrl = realtimeBaseUrl ?? baseUrl.replaceAll('/api', '') + '/api/realtime/';
    final cleanBaseUrl = baseUrl.endsWith('/') ? baseUrl : '$baseUrl/';

    database = DatabaseService(
      client: client,
      apiKey: apiKey,
      baseUrl: rUrl,
      databaseId: databaseId ?? '',
    );

    calls = CallsService(
      client: client,
      apiKey: apiKey,
      baseUrl: cleanBaseUrl,
    );

    ai = AiService(
      client: client,
      apiKey: apiKey,
      baseUrl: cleanBaseUrl,
    );

    email = EmailService(
      client: client,
      apiKey: apiKey,
      baseUrl: cleanBaseUrl,
    );

    sms = SmsService(
      client: client,
      apiKey: apiKey,
      baseUrl: cleanBaseUrl,
    );

    storage = StorageService(
      client: client,
      apiKey: apiKey,
      baseUrl: cleanBaseUrl,
    );
  }
}

File: pingdart-php/src/PingDartSDK.php
<?php

namespace PingDart\SDK;

use GuzzleHttp\Client;
use PingDart\SDK\Services\DatabaseService;
use PingDart\SDK\Services\WhatsAppService;
use PingDart\SDK\Services\SmsService;
use PingDart\SDK\Services\EmailService;
use PingDart\SDK\Services\AiService;
use PingDart\SDK\Services\CallsService;
use PingDart\SDK\Services\StorageService;

class PingDartSDK {
    public $database;
    public $whatsapp;
    public $sms;
    public $email;
    public $ai;
    public $calls;
    public $storage;

    private $apiKey;
    private $databaseId;
    private $baseUrl;
    private $realtimeBaseUrl;

    public function __construct(array $config) {
        $this->apiKey = $config['apiKey'] ?? null;
        if (!$this->apiKey) {
            throw new \Exception("PingDart SDK: API Key is required");
        }

        $this->databaseId = $config['databaseId'] ?? null;
        $this->baseUrl = rtrim($config['baseUrl'] ?? 'https://cloudapi.pingdart.com/api', '/');
        $this->realtimeBaseUrl = $config['realtimeBaseUrl'] ?? str_replace('/api', '', $this->baseUrl) . '/api/realtime/';

        $client = new Client([
            'base_uri' => $this->baseUrl . '/',
            'headers' => [
                'x-api-key' => $this->apiKey,
                'Content-Type' => 'application/json',
                'X-SDK-Source' => 'PingDart-PHP-SDK'
            ]
        ]);

        $realtimeClient = new Client([
            'base_uri' => $this->realtimeBaseUrl,
            'headers' => [
                'x-api-key' => $this->apiKey,
                'Content-Type' => 'application/json',
                'X-SDK-Source' => 'PingDart-PHP-SDK'
            ]
        ]);

        $this->database = new DatabaseService($realtimeClient, $this->databaseId);
        $this->whatsapp = new WhatsAppService($client);
        $this->sms = new SmsService($client);
        $this->email = new EmailService($client);
        $this->ai = new AiService($client);
        $this->calls = new CallsService($client);
        $this->storage = new StorageService($client);
    }
}
